jaxrs/WFLY-13585_Galleon_layer_jsonp_support.adoc

```diff
@@ -0,0 +1,113 @@
+= Galleon layer for JSON-P support
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Jakarta JSON Processing (JSON-P) is a Java API to process (e.g. parse, generate, transform and query) JSON documents. It produces and consumes JSON in a streaming fashion (similar to StAX API for XML) and allows to build a Java object model for JSON using API classes (similar to DOM API for XML).
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] which will be responsible for providing the JSON-P API and Implementation.
+
+The Galleon packages that provision JSON-P are:
+
+ * `javax.json.api`: It provisions the JBoss Modules module for JSON-P API
+ * `org.glassfish.jakarta.json`: It provisions the JBoss Modules module for JSON-P implementation
+
+Since the JSON-P API Jboss Modules module depends on JSON-P Impl module, the implementation is always provisioned as soon as the API is present. The JSON-P API module is provisioned always by the `base-server` Galleon layer; there are modules provisioned by the `base-server` that have a hard dependency on the JSON-P modules, e.g. `org.jboss.logmanager` and `org.wildfly.security.elytron-private`.
+
+This means we create a JSON-P layer to ensure that JSON-P modules will be always provisioned for any layer that wants to depend on JSON-P layer explicitly, making them more robust to possible `base-server` changes.
+
+The most relevant layer for this use case is the `jaxrs` Galleon layer, however, this layer does not need to depend directly on JSON-P layer because the JSON-P API module is among its own dependencies.
+
+The following is a configuration example of the WildFly bootable JAR Maven plugin that provisions the `jaxrs-server` Galleon layer which already includes the JSON-P API and Implementation:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>jaxrs-server</layer>
```

Comment:
Sorry because I might have got it wrong. Why do we use an existing layer (I am aware it provides the JSON-P AI and impl.) instead of just using the new module? BTW what would it be the new module name?

Reply:
I'm not sure what do you mean here with "module", could you elaborate it a bit more?

We could replace in the example the `jaxrs-server` Galleon layer by `jsonp`, however, it does not look as useful case, for this reason I used `jaxrs-server` and an example. Since `jsonp` is always provisioned,` base-server` Galleon layer, which is the smaller Galleon layer, would be enough to show an example, however as I said, `jaxrs-server` sounds more relevant.

Reply:
> I'm not sure what do you mean here with "module", could you elaborate it a bit more?

Sorry, I used the wrong word for "layer". 

> We could replace in the example the jaxrs-server Galleon layer by jsonp, however, it does not look as useful case, for this reason I used jaxrs-server and an example. Since jsonp is always provisioned, base-server Galleon layer, which is the smaller Galleon layer, would be enough to show an example, however as I said, jaxrs-server sounds more relevant.

Agree, thanks.